File: app/src/main/java/com/eccentricyan/pohe/common/command/ViewBindingAdapter.java
package com.eccentricyan.pohe.common.command;

import android.databinding.BindingAdapter;
import android.support.design.widget.AppBarLayout;
import android.support.v4.widget.SwipeRefreshLayout;
import android.support.v7.widget.RecyclerView;
import android.text.TextUtils;
import android.view.View;
import android.webkit.WebView;
import android.widget.ImageView;

import com.bumptech.glide.Glide;
import com.eccentricyan.pohe.common.listener.OnRcvScrollListener;

/**
 * Created by shiyanhui on 2017/04/15.
 */

public class ViewBindingAdapter {
    @BindingAdapter({"clickCommand"})
    public static void clickCommand(View view, final ReplyCommand clickCommand) {
        view.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (clickCommand != null) {
                    clickCommand.execute();
                }
            }
        });
    }
    @BindingAdapter({"onLoadMoreCommand"})
    public static void onLoadMoreCommand(RecyclerView recyclerView, final ReplyCommand
            onLoadMoreCommand) {
        recyclerView.addOnScrollListener(new OnRcvScrollListener() {
            @Override
            public void onBottom() {
                if (onLoadMoreCommand != null) {
                    onLoadMoreCommand.execute();
                }
            }

            @Override
            public int setRestItem() {
                return 20;
            }
        });
    }

    @BindingAdapter({"imageUrl"})
    public static void loadImage(ImageView view, String imageUrl) {
        if (TextUtils.isEmpty(imageUrl)) {
//            Glide.with(view.getContext()).load(SITE_URL + "/images/face_male.jpg").fitCenter().into(view);
        } else {
            Glide.with(view.getContext())
                    .load(imageUrl)
                    //                .placeholder(R.drawable.placeholder)
                    .into(view);
        }
    }

    @BindingAdapter("addOnOffsetChangedListener")
    public static void initToolbar(AppBarLayout appBarLayout, AppBarLayout
            .OnOffsetChangedListener listener) {
        appBarLayout.addOnOffsetChangedListener(listener);
    }

    @BindingAdapter({"isRefresh"})
    public static void setRefresh(SwipeRefreshLayout swipeRefreshLayout, boolean isRefreshing) {
        swipeRefreshLayout.setRefreshing(isRefreshing);
    }

    @BindingAdapter({"onRefreshCommand"})
    public static void onRefreshCommand(SwipeRefreshLayout swipeRefreshLayout, final ReplyCommand
            onRefreshCommand) {
        swipeRefreshLayout.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
            @Override
            public void onRefresh() {
                if (onRefreshCommand != null) {
                    onRefreshCommand.execute();
                }
            }
        });
    }

    @BindingAdapter({"render"})
    public static void loadHtml(WebView webView, final String html) {
        if (!TextUtils.isEmpty(html)) {
            webView.loadDataWithBaseURL(null, html, "text/html", "UTF-8", null);
        }
    }

}

File: app/src/main/java/com/eccentricyan/pohe/common/command/ViewBindingAdapter.kt
package com.eccentricyan.pohe.common.command

import android.databinding.BindingAdapter
import android.support.design.widget.AppBarLayout
import android.support.v4.widget.SwipeRefreshLayout
import android.support.v7.widget.RecyclerView
import android.text.TextUtils
import android.view.View
import android.webkit.WebView
import android.widget.ImageView

import com.bumptech.glide.Glide
import com.eccentricyan.pohe.common.listener.OnRcvScrollListener

/**
 * Created by shiyanhui on 2017/04/15.
 */

object ViewBindingAdapter {
    @JvmStatic
    @BindingAdapter("clickCommand")
    fun clickCommand(view: View, clickCommand: ReplyCommand<*>?) {
        view.setOnClickListener {
            clickCommand?.execute()
        }
    }

    @JvmStatic
    @BindingAdapter("onLoadMoreCommand")
    fun onLoadMoreCommand(recyclerView: RecyclerView, onLoadMoreCommand: ReplyCommand<*>?) {
        recyclerView.addOnScrollListener(object : OnRcvScrollListener() {
            override fun onBottom() {
                onLoadMoreCommand?.execute()
            }

            override fun setRestItem(): Int {
                return 20
            }
        })
    }

    @JvmStatic
    @BindingAdapter("imageUrl")
    fun loadImage(view: ImageView, imageUrl: String?) {
        if (imageUrl.isNullOrEmpty())
            Glide.with(view.context)
                        .load(imageUrl)
                        .into(view)
    }

    @JvmStatic
    @BindingAdapter("addOnOffsetChangedListener")
    fun initToolbar(appBarLayout: AppBarLayout, listener: AppBarLayout.OnOffsetChangedListener) {
        appBarLayout.addOnOffsetChangedListener(listener)
    }

    @JvmStatic
    @BindingAdapter("isRefresh")
    fun setRefresh(swipeRefreshLayout: SwipeRefreshLayout, isRefreshing: Boolean) {
        swipeRefreshLayout.isRefreshing = isRefreshing
    }

    @JvmStatic
    @BindingAdapter("onRefreshCommand")
    fun onRefreshCommand(swipeRefreshLayout: SwipeRefreshLayout, onRefreshCommand: ReplyCommand<*>?) {
        swipeRefreshLayout.setOnRefreshListener {
            onRefreshCommand?.execute()
        }
    }

    @JvmStatic
    @BindingAdapter("render")
    fun loadHtml(webView: WebView, html: String) {
        if (!TextUtils.isEmpty(html)) {
            webView.loadDataWithBaseURL(null, html, "text/html", "UTF-8", null)
        }
    }

}
